### ISSUE TITLE
Building for iOS fails: ld: framework not found Flutter

### ISSUE THREAD

**Issue by QJune** [flutter/flutter#43591]
```
Xcode's output:
↳
    ld: framework not found Flutter
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build systemnote: Planning buildnote: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 11 Pro Max.
```

**Comment by juneqiu** [comment]
my xcode version is 11.1
it's no problem with xcode 10

**Comment by BondarenkoStas** [comment]
@juneqiu  Please check https://stackoverflow.com/a/54860339/12268368.

**Comment by jmagman** [comment]
What exact command produced that?  Flutter is not supported in Release mode on the simulator.

**Comment by juneqiu** [comment]
> What exact command produced that? Flutter is not supported in Release mode on the simulator.

flutter run. it can run in android stadio

**Comment by jmagman** [comment]
> > What exact command produced that? Flutter is not supported in Release mode on the simulator.
> 
> flutter run. it can run in android stadio

`flutter run` defaults to release mode.  You can target a real device with release, or the simulator with debug.
```
flutter run --debug
```
Duplicate of https://github.com/flutter/flutter/issues/11754.

Let's keep this open, I think we should detect the simulator is being targeted in release mode and short-circuit with a better error message.

**Comment by juneqiu** [comment]
> > > What exact command produced that? Flutter is not supported in Release mode on the simulator.
> > 
> > 
> > flutter run. it can run in android stadio
> 
> `flutter run` defaults to release mode. You can target a real device with release, or the simulator with debug.
> 
> ```
> flutter run --debug
> ```
> 
> Duplicate of #11754.
> 
> Let's keep this open, I think we should detect the simulator is being targeted in release mode and short-circuit with a better error message.

`qiujundeMacBook-Pro:flutter_shop qiujun$ flutter run --debug
Launching lib/main.dart on iPhone 11 in debug mode...
Running Xcode build...                                                  
 ├─Assembling Flutter resources...                           6.9s
 └─Compiling, linking and signing...                         1.0s
Xcode build done.                                            9.7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    ld: framework not found Flutter
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build systemnote: Planning buildnote: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 11.`

**Comment by juneqiu** [comment]
> > > What exact command produced that? Flutter is not supported in Release mode on the simulator.
> > 
> > 
> > flutter run. it can run in android stadio
> 
> `flutter run` defaults to release mode. You can target a real device with release, or the simulator with debug.
> 
> ```
> flutter run --debug
> ```
> 
> Duplicate of #11754.
> 
> Let's keep this open, I think we should detect the simulator is being targeted in release mode and short-circuit with a better error message.

failed to build

**Comment by jmagman** [comment]
Do you have any plugins?
Does `ios/Runner.xcworkspace` successfully build in Xcode?

**Comment by juneqiu** [comment]
> Do you have any plugins?
> Does `ios/Runner.xcworkspace` successfully build in Xcode?

build in xcode with the problem,but i run flutter doctor can find flutter


ld: framework not found Flutter
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by jmagman** [comment]
1. What's the output of `flutter doctor -v`?
2. Do you have any plugins?
3. When you open in Xcode do you see Flutter.framework in your project?
![Screen Shot 2019-10-28 at 6 49 26 PM](https://user-images.githubusercontent.com/682784/67730951-cd47ae80-f9b3-11e9-8dd5-4284c8e8b2c2.png)

**Comment by juneqiu** [comment]
> 1. What's the output of `flutter doctor -v`?
> 2. Do you have any plugins?
> 3. When you open in Xcode do you see Flutter.framework in your project?
>    ![Screen Shot 2019-10-28 at 6 49 26 PM](https://user-images.githubusercontent.com/682784/67730951-cd47ae80-f9b3-11e9-8dd5-4284c8e8b2c2.png)

thank you!I have resolve the problem by install flutter again!

**Comment by juneqiu** [comment]
> @juneqiu Please check https://stackoverflow.com/a/54860339/12268368.

thanks,i have resolve it by install flutter again

**Comment by jmagman** [comment]
You resolved the issue by reinstalling flutter again?  Do you know what version you were on before?
I'm concerned that framework ever disappeared.

**Comment by juneqiu** [comment]
Yes,it can run with iOS simulators ,when i reinstalling flutter.  I have forgotten  the old version. The old version was installed about three or four months ago.

**Comment by Skullper** [comment]
I have faced the same issue after re-installing XCode. To fix this I've just upgraded Flutter `flutter upgrade` and run `pod install`. Hope it'll help someone

**Comment by QJune** [comment]
> I have faced the same issue after re-installing XCode. To fix this I've just upgraded Flutter `flutter upgrade` and run `pod install`. Hope it'll help someone

yes,My flutter version was too low at the time

**Comment by srjmalla** [comment]
This fixed the issue for me

--    flutter channel stable

--   flutter upgrade

**Comment by roshan-alexander** [comment]
I just ran into this issue after removing a dependency package 'fluttertoast' from the pubspec file. Tried running `flutter pub get` and then ran `flutter run --debug` and received the following output.

```
Launching lib/main.dart on iPhone 11 Pro in debug mode...
Running pod install...                                              3.3s
Running Xcode build...                                                  
                                                   
 └─Compiling, linking and signing...                         2.5s
Xcode build done.                                           21.7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/ra/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.15.1/darwin/Classes/AudioplayersPlugin.m:88:37: warning: incompatible pointer types sending 'FlutterEngine *' to parameter of type 'NSObject<FlutterBinaryMessenger> *
    _Nonnull' [-Wincompatible-pointer-types]
                        binaryMessenger:_headlessEngine];
                                        ^~~~~~~~~~~~~~~
    In module 'Flutter' imported from /Users/ra/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.15.1/darwin/Classes/AudioplayersPlugin.h:2:
    /Users/ra/Documents/Workspace/Flutter Projects/test/ios/Flutter/Flutter.framework/Headers/FlutterChannels.h:178:74: note: passing argument to parameter 'messenger' here
                          binaryMessenger:(NSObject<FlutterBinaryMessenger>*)messenger;
                                                                             ^
    1 warning generated.
    /Users/ra/Documents/Workspace/Flutter Projects/test/ios/Runner/AppDelegate.swift:13:58: warning: conditional cast from 'AppDelegate' to 'UNUserNotificationCenterDelegate' always succeeds
          UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
                                                             ^
    ld: framework not found fluttertoast
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 11 Pro.

```
